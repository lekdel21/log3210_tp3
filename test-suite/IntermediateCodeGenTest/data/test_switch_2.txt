num b;
enum A {
    B, C, D
}
A a;

a = D;

switch(a) {
    case B:
    case D:
        b = 6;
        break;
    case C:
        b =  10;
        break;
}