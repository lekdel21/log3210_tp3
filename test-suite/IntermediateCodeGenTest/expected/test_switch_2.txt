a = 2
_L1
if a != 0 goto _L2
goto _L3
_L2
if a != 2 goto _L4
_L3
b = 6
goto _L0
_L4
if a != 1 goto _L0
b = 10
goto _L0
_L0
